source/DSCResources/DSC_CMCollections/DSC_CMCollections.psm1

```diff
@@ -0,0 +1,685 @@
+                        {
+                            $setSchedule = $true
+                        }
+                    }
+                }
+                else
+                {
+                    $setSchedule = $true
+                }
+            }
+
+            if ($setSchedule)
+            {
+                $buildingParams += @{
+                    RefreshSchedule = $desiredRefreshSchedule
+                }
+            }
+
+            if ($buildingParams.Count -gt 1)
+            {
+                Set-CMCollection @buildingParams
+            }
+
+            if (-not [string]::IsNullOrEmpty($ExcludeMembership))
+            {
+                foreach ($member in $ExcludeMembership)
+                {
+                    $excludeRule = @{}
+
+                    if (($null -eq $state.ExcludeMembership) -or ($state.ExcludeMembership -notcontains $member))
+                    {
+                        $excludeRule = @{
+                            CollectionName = $CollectionName
+                            ExcludeCollectionName = $member
+                        }
+                        
+                        Write-Verbose -Message ($script:localizedData.ExcludeMemberRule -f $CollectionName, $member)
+
+                        if ($CollectionType -eq 'User')
+                        {
+                            Add-CMUserCollectionExcludeMembershipRule @excludeRule
+                        }
+                        else
+                        {
+                            Add-CMDeviceCollectionExcludeMembershipRule @excludeRule
+                        }
+                    }
+                }
+            }
+
+            if (-not [string]::IsNullOrEmpty($DirectMembership))
+            {
+                foreach ($member in $DirectMembership)
+                {
+                    $directRule = @{}
+
+                    if (($null -eq $state.DirectMembership) -or ($state.DirectMembership -notcontains $member))
+                    {
+                        $directRule = @{
+                            CollectionName = $CollectionName
+                            ResourceId     = $member
+                        }
+                        
+                        Write-Verbose -Message ($script:localizedData.DirectMemberRule -f $CollectionName, $member)
+
+                        if ($CollectionType -eq 'User')
+                        {
+                            Add-CMUserCollectionDirectMembershipRule @directRule
+                        }
+                        else
+                        {
+                            Add-CMDeviceCollectionDirectMembershipRule @directRule
+                        }
+                    }
+                }
+            }
+
+            if (-not [string]::IsNullOrEmpty($QueryRules))
+            {
+                $rules = @()
+                foreach ($queryRule in $QueryRules)
+                {
+                    $rules += (
```

Comment:
Do you need to make this into a hashtable. In the next section can you call the properties?

Reply:
tested and it is unable to handle the cim instance.  a conversion would need to be done to be able to use this as specified.

Reply:
I made hashtables as the code lines are long without it.